test/integration/targets/docker_swarm_service/tasks/tests/options.yml

```diff
@@ -0,0 +1,1696 @@
+    network_name_2: "{{ name_prefix ~ '-network-2' }}"
+    config_name_1: "{{ name_prefix ~ '-configs-1' }}"
+    config_name_2: "{{ name_prefix ~ '-configs-2' }}"
+    volume_name_1: "{{ name_prefix ~ '-volume-1' }}"
+    volume_name_2: "{{ name_prefix ~ '-volume-2' }}"
+    secret_name_1: "{{ name_prefix ~ '-secret-1' }}"
+    secret_name_2: "{{ name_prefix ~ '-secret-2' }}"
+
+- name: Registering container name
+  set_fact:
+    service_names: "{{ service_names }} + [service_name]"
+    network_names: "{{ network_names }} + [network_name_1, network_name_2]"
+    config_names: "{{ config_names }} + [config_name_1, config_name_2]"
+    volume_names: "{{ volume_names }} + [volume_name_1, volume_name_2]"
+    secret_names: "{{ secret_names }} + [secret_name_1, secret_name_2]"
+
+- name: Create a Swarm cluster
+  docker_swarm:
+    advertise_addr: "{{ ansible_default_ipv4.address }}"
+    state: present
+
+- docker_config:
+    name: "{{ config_name_1 }}"
+    data: "hello"
+    state: present
+  register: "config_result_1"
+
+- docker_config:
+    name: "{{ config_name_2 }}"
+    data: "test"
+    state: present
+  register: "config_result_2"
+
+- docker_secret:
+    name: "{{ secret_name_1 }}"
+    data: "secret1"
+    state: "present"
+  register: "secret_result_1"
+
+- docker_secret:
+    name: "{{ secret_name_2 }}"
+    data: "secret2"
+    state: "present"
+  register: "secret_result_2"
+
+- docker_network:
+    name: "{{ network_name }}"
+    driver: "overlay"
+    state: present
+  loop:
+    - "{{ network_name_1 }}"
+    - "{{ network_name_2 }}"
+  loop_control:
+    loop_var: network_name
+
+- docker_volume:
+    name: "{{ volume_name }}"
+    state: present
+  loop:
+    - "{{ volume_name_1 }}"
+    - "{{ volume_name_2 }}"
+  loop_control:
+    loop_var: volume_name
+
+####################################################################
+## args ############################################################
+####################################################################
+
+- name: args
+  docker_swarm_service:
+    name: "{{ service_name }}"
+    image: alpine:3.8
+    args:
+      - sleep
+      - "3600"
+  register: args_1
+
+- name: args (idempotency)
+  docker_swarm_service:
+    name: "{{ service_name }}"
+    image: alpine:3.8
+    args:
+      - sleep
+      - "3600"
+  register: args_2
+
+- name: args (change)
+  docker_swarm_service:
+    name: "{{ service_name }}"
+    image: alpine:3.8
+    args:
+      - sleep
+      - "3400"
+  register: args_3
+
+- name: cleanup
+  docker_swarm_service:
+    name: "{{ service_name }}"
+    state: absent
+  diff: no
+
+- assert:
+    that:
+      - args_1 is changed
+      - args_2 is not changed
+      - args_3 is changed
+
+####################################################################
+## configs #########################################################
+####################################################################
+
+# FIXME: Broken until #50606 is merged
```

Comment:
I think this needs to be uncommented.

Reply:
Sure. Just wanted to show that the tests pass.

Reply:
Do you want me to uncomment tests for publish to as they should be fixed by #50882?
https://github.com/hannseman/ansible/blob/a98e69984701498305903f776938a00f999e08f3/test/integration/targets/docker_swarm_service/tasks/tests/options.yml#L955

Then the only FIXME left will be the one for `endpoint_mode` and I haven't gotten time yet to find the cause of why it is marked as changed:
https://github.com/hannseman/ansible/blob/a98e69984701498305903f776938a00f999e08f3/test/integration/targets/docker_swarm_service/tasks/tests/options.yml#L423

Reply:
I'd rather get this merged first; then you can rebase #50882 to include tests for it.